frontend/packages/shared/defi/lib/Apollo.ts

```diff
@@ -0,0 +1,28 @@
+import { ApiPromise } from "@polkadot/api";
+import { Asset } from "./assets/Asset";
+import BigNumber from "bignumber.js";
+
+export class Apollo {
+  protected __api: ApiPromise;
+
+  constructor(api: ApiPromise) {
+    this.__api = api;
+  }
+
+  public async getPrice(assets: Asset[]): Promise<Record<string, BigNumber>> {
+    let priceMap: Record<string, BigNumber> = {};
+    try {
+      for (const asset of assets) {
+        const assetId: string = asset.getPicassoAssetId() as string;
+        let prices = await this.__api.query.oracle.prices(assetId);
+        const price = new BigNumber(prices.price.toString());
+        // @ts-ignore
+        priceMap[assetId] = price;
```

Comment:
While this looks good, I think we need to subscribe to price changes from Apollo. 

If you don't plan to change that now, let's add a todo.